File: GoOfficialClient/Producer/producer.go
package main

import (
	"context"
	"fmt"
	"log"
	"sync"
	"time"

	"github.com/apache/pulsar-client-go/pulsar"
)

func main() {
	client, err := pulsar.NewClient(pulsar.ClientOptions{
		URL:               "pulsar://127.0.0.1:6650",
		OperationTimeout:  30 * time.Second,
		ConnectionTimeout: 30 * time.Second,
	})
	if err != nil {
		log.Fatalf("Could not instantiate Pulsar client: %v", err)
	}

	producer, err := client.CreateProducer(pulsar.ProducerOptions{
		Topic: "my-topic",
	})

	if err != nil {
		fmt.Println("Failed to publish message", err)
		return
	}

	payload := []byte("hello")
	ctx := context.Background()

	n := 1000000

	wg := &sync.WaitGroup{}

	//warm up
	for j := 1; j <= n/10000; j++ {
		for i := 0; i < 1000; i++ {
			wg.Add(1)
			producer.SendAsync(ctx, &pulsar.ProducerMessage{
				Payload: payload,
			}, func(mid pulsar.MessageID, msg *pulsar.ProducerMessage, err error) {
				defer wg.Done()
			})
		}
		wg.Wait()
	}

	start := time.Now()
	for j := 1; j <= n/1000; j++ {
		for i := 0; i < 1000; i++ {
			wg.Add(1)
			producer.SendAsync(ctx, &pulsar.ProducerMessage{
				Payload: payload,
			}, func(mid pulsar.MessageID, msg *pulsar.ProducerMessage, err error) {
				defer wg.Done()
			})
		}
		wg.Wait()
		if j%10 == 0 {
			fmt.Println("Published message", j*1000)
		}
	}

	elapsed := time.Since(start)
	speed := float64(n) / (elapsed.Seconds() * 1000)

	fmt.Printf("Loop executed in %s, speed: %.2fK iterations/second\n", elapsed, speed)
}

File: NodeOfficialClient/producer.js
const Pulsar = require('pulsar-client');

(async () => {
  const client = new Pulsar.Client({
    serviceUrl: "pulsar://127.0.0.1:6650",
    log: (level, file, line, message) => level>=Pulsar.LogLevel.WARN && console.log(`[${level}] ${file}:${line} ${message}`)
  });

  const producer = await client.createProducer({
    topic: 'my-topic', 
    blockIfQueueFull: true
  });
  
  const data = Buffer.alloc(750)

  const n = 10000
  let i = 0;
  // warmup
  while(i < n/10) {
    producer.send({
        data
    })
    i ++
  }
  await producer.send({data})

  let start = performance.now()
  let j = 0;
  while(j < n-1) {
    producer.send({
        data
    })
    j ++
    if (j  % (n/100) == 0)
    {
        console.log("Sent " + j  + " messages");
    }
  }
  await producer.send({data})
  let elapsedMs = performance.now() - start
  console.log("Time taken: " + elapsedMs + "ms.Speed: " + n / elapsedMs  + "K msg/s")
  
})();